src/py/aspen/app/views/tests/test_auspice_redirect.py

```diff
@@ -0,0 +1,58 @@
+import json
+
+from aspen.test_infra.models.phylo_tree import phylotree_factory
+from aspen.test_infra.models.sample import sample_factory
+from aspen.test_infra.models.usergroup import group_factory, user_factory
+
+
+def test_auspice_redirect_view(
+    session, app, client, s3_resource, s3_client, test_data_dir
+):
+    group = group_factory()
+    session.add(group)
+    session.commit()
+
+    user = user_factory(group)
+
+    sample_1 = sample_factory(
+        group,
+        private_identifier="private_identifer_1",
+        public_identifier="public_identifier_1",
+    )
+    sample_2 = sample_factory(
+        group,
+        private_identifier="private_identifer_2",
+        public_identifier="public_identifier_2",
+    )
+
+    session.add(sample_1)
+    session.add(sample_2)
+    session.commit()
+
+    phylo_tree = phylotree_factory([sample_1, sample_2])
+    session.add(phylo_tree)
+    session.commit()
+
+    # We need to create the bucket since this is all in Moto's 'virtual' AWS account
+    s3_resource.create_bucket(Bucket=phylo_tree.s3_bucket)
+
+    json_test_file = test_data_dir / "ncov_aspen.json"
+    with json_test_file.open() as fh:
+        test_json = json.dumps(json.load(fh))
+
+    s3_client.put_object(
+        Bucket=phylo_tree.s3_bucket, Key=phylo_tree.s3_key, Body=test_json
+    )
+
+    with client.session_transaction() as sess:
+        sess["profile"] = {"name": user.name, "user_id": user.auth0_user_id}
+
+    res = client.get(f"/api/auspice/view/{phylo_tree.id}")
+    presigned_url_sections = json.loads(res.data)["url"].split("/")
```

Comment:
You could just do `res.json["url"].split("/")`

Reply:
unfortunately you cannot do this (`res.json` is `None`)

Reply:
Wait that doesn't make sense.  If it's json it ought to be available as `.json`

Reply:
oh wait, it's [slightly different](https://flask.palletsprojects.com/en/1.1.x/testing/#testing-json-apis) than python requests.  Maybe it's `rv.get_json()`?

Reply:
let me try that

Reply:
that also does not work :( it's wierd that it's so hard to get the json out of this response object..

Reply:
So it seems to work if you tweak the result value in the handler.